RoaringBitmap/src/main/java/org/roaringbitmap/FastAggregation.java

```diff
@@ -262,12 +264,76 @@ public static RoaringBitmap naive_and(RoaringBitmap... bitmaps) {
       return new RoaringBitmap();
     }
     RoaringBitmap answer = bitmaps[0].clone();
-    for (int k = 1; k < bitmaps.length; ++k) {
+    for (int k = 1; k < bitmaps.length && !answer.isEmpty(); ++k) {
       answer.and(bitmaps[k]);
     }
     return answer;
   }
 
+  /**
+   * Computes the intersection by first intersecting the keys, avoids
+   * materialising containers.
+   *
+   * @param bitmaps the inputs
+   * @return the intersection of the bitmaps
+   */
+  public static RoaringBitmap workShyAnd(RoaringBitmap... bitmaps) {
+    long[] words = new long[1024];
```

Comment:
why is this hardcoded to 1024.

Reply:
What @richardstartin  is doing is basically allocating the equivalent of a BitmapContainer. The 1024 comes from the Roaring format and is indeed a constant.

Reply:
(E.g., it has the number of bits needed to represent a 16-bit space.) (Note that, here, it is not used for the low 16 bits.)